plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -162,29 +162,19 @@ public void checkVersionSpecificUpdateCenter() {
                 System.out.println(
                         "Unable to check if version specific update center for Jenkins version " + jenkinsVersion);
             }
-
         }
-
     }
 
-
-    public void writeFailedPluginsToFile() {
-        try (
-          FileOutputStream fileOutputStream = new FileOutputStream("failedplugins.txt");
-          Writer fstream = new OutputStreamWriter(fileOutputStream, StandardCharsets.UTF_8)
-        ) {
-                if (failedPlugins.size() > 0) {
-                    System.out.println("Some plugins failed to download: ");
-                    for (Plugin plugin : failedPlugins) {
-                        String failedPluginName = plugin.getName();
-                        System.out.println(failedPluginName);
-                        fstream.write(failedPluginName + "\n");
-                    }
-                }
-            } catch (IOException e) {
-                e.printStackTrace();
+    public void outputFailedPlugins() {
+        if (failedPlugins.size() > 0) {
+            System.out.println("Some plugins failed to download: ");
+            for (Plugin plugin : failedPlugins) {
```

Comment:
This might be slightly simpler:
```
        failedPlugins.stream()
                .map(Plugin::getName)
                .forEach(System.out::println);
```